client/state/support/reducer.js

```diff
@@ -1,39 +1,61 @@
 /** @format */
 
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+
 /**
  * Internal dependencies
  */
-import { createReducer, combineReducers } from 'state/utils';
+import { combineReducers } from 'state/utils';
 import {
-	SUPPORT_SESSION_ACTIVATE,
-	SUPPORT_SESSION_EXPIRE,
+	SUPPORT_SESSION_TRANSITION,
 	SUPPORT_USER_ACTIVATE,
 	SERIALIZE,
 	DESERIALIZE,
 } from 'state/action-types';
 
+const debug = debugFactory( 'calypso:state:support:actions' );
+
+export const SESSION_NONE = 'none';
+export const SESSION_ACTIVE = 'active';
+export const SESSION_EXPIRED = 'expired';
+
 export const SUPPORT_SESSION_STATE = {
 	NONE: 'none',
 	ACTIVE: 'active',
 	EXPIRED: 'expired',
 };
 
-export const supportSession = createReducer( SUPPORT_SESSION_STATE.NONE, {
-	[SUPPORT_SESSION_ACTIVATE]( state ) {
-		return state === SUPPORT_SESSION_STATE.NONE ? SUPPORT_SESSION_STATE.ACTIVE : state;
-	},
-	[SUPPORT_SESSION_EXPIRE]( state ) {
-		return state === SUPPORT_SESSION_STATE.ACTIVE ? SUPPORT_SESSION_STATE.EXPIRED : state;
-	},
-	// Use custom serialization so we're able to relay supportSession state from server to client.
-	// Otherwise, only the default state is used when deserializing redux state from the server.
-	[SERIALIZE]() {
-		return null;
-	},
-	[DESERIALIZE]( state ) {
-		return state === null ? SUPPORT_SESSION_STATE.NONE : state;
-	},
-} );
+function supportSession( state = SESSION_NONE, { type, nextState } ) {
+	switch ( type ) {
+		case SUPPORT_SESSION_TRANSITION:
+			if (
+				( state === SESSION_NONE && nextState === SESSION_ACTIVE ) ||
+				( state === SESSION_ACTIVE && nextState === SESSION_EXPIRED )
+			) {
+				return nextState;
+			}
+
+			debug( `invalid support session transition from '${ state }' to '${ nextState }'` );
+			return state;
+
+		// Support custom persistence so the client is willing to consume this state from the server.
+		// We don't actually want to persist this state, so serialize to null.
+		// TODO: Consider way to relay non-persisted state from the server and remove this hack.
+		case SERIALIZE:
+			return null;
+		case DESERIALIZE:
+			return state === null ? SESSION_NONE : state;
+
+		default:
+			return state;
+	}
+}
+supportSession.hasCustomPersistence = true;
+
+export { supportSession };
```

Comment:
It should be fine. It's just exporting the identified thing under the same name. Check out how Babel transpiles it [here](https://babeljs.io/repl/#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=KYDwDg9gTgLgBAYwgOwM7wEbHQOQK4C2WUcAvHAIwDcAUDQGZ7IIwCWKcW6AYky-8gAUASjgBvGnDhRgMPFGSdsMfEWBRaAXxqhIscUp582HTVSA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=6.26.0&envVersion=). Your comment made me want to check the spec, and I believe this syntax is `export ExportClause;` [here](https://www.ecma-international.org/ecma-262/9.0/index.html#sec-exports).

It's possible to export the function declaration, but since we're adding a property to it, it seemed clearer to declare, modify, and export separately and in that order.

Reply:
> It's possible to export the function declaration, but since we're adding a property to it, it seemed clearer to declare, modify, and export separately and in that order.

Doing what you are doing is fine. I'm not sure it's clearer one way or the other - that is, I doubt you'd find that in practice the `export` placement has any impact on the clarity of the mutation. I may be wrong!

Reply:
When exporting reducers, we always use a default export:
```js
export default supportSession;
```
```js
import support from 'state/support/reducer';

combineReducers( { ..., support, ... } );
```
Nothing but a convention, but still advisable to follow it.

Reply:
@jsnajdr This named export is just there to expose the reducer for unit test. There is a default export further down in the the module.